cmd/goal/clerk.go

```diff
@@ -1253,6 +1257,34 @@ var dryrunRemoteCmd = &cobra.Command{
 	},
 }
 
+var simulateCmd = &cobra.Command{
+	Use:   "simulate",
+	Short: "Simulate a transaction or transaction group offline",
+	Long:  `Simulate a transaction or transaction group offline under various conditions and verbosity.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		data, err := readFile(txFilename)
+		if err != nil {
+			reportErrorf(fileReadError, txFilename, err)
+		}
+
+		dataDir := datadir.EnsureSingleDataDir()
```

Comment:
In dryrun, it looks like we also try to decode the txns and report a decoding error on the client side if this fails: https://github.com/algorand/go-algorand/blob/3e1772908f3643ceb818b3afca18da07e6f606bc/cmd/goal/clerk.go#L1126

Do we want to do something similar here, or just fall back on the server side error?

Reply:
I think we send the transaction to server, then by following line
https://github.com/algorand/go-algorand/blob/master/daemon/algod/api/server/v2/handlers.go#L955

The decoding error should fail on server side.